File: PracticaExamen/Logica/ListaClientes.vb
Imports AccesoDatos
Imports ObjetosTransferencia
Public Class ListaClientes
    Private lista As List(Of Cliente)

    Sub New()
        lista = New List(Of Cliente)
    End Sub

    Sub agregar(c As Cliente)
        lista.Add(c)
    End Sub

    Function obtener(i As Integer) As Cliente
        Return lista.Item(i)
    End Function

    Function obtener() As List(Of Cliente)
        Return Me.lista
    End Function

    Sub cargar()
        Dim daoC As DAOCliente = New DAOCliente()

        For Each c As TOCliente In daoC.cargar
            Me.agregar(New Cliente(c.Cedula, c.Nombre))
        Next

    End Sub
End Class

File: Practicav2/Logica/ListaClientes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AccesoDatos;
using ObjetosTransferencia;

namespace Logica
{
    public class ListaClientes
    {
        public List<Cliente> lista { get; set; }

        public ListaClientes() {
            lista = new List<Cliente>();
        }

        public void cargar() {
            DAOCliente daoC = new DAOCliente();
            TOListaClientes toL = daoC.selectClientes();

            foreach (TOCliente item in toL.lista)
            {
                lista.Add(new Cliente(item.Cedula, item.Nombre));
            }
        }

        public void refresh() {
            lista = new List<Cliente>();
            cargar();
        }
    }
}
